### PR TITLE
Soft float

### LINKED ISSUE #586: Add support for logb, logbf, logbl
https://en.cppreference.com/w/c/numeric/math/logb

### PR DESCRIPTION
**Pull request by tgymnich**
fixes  #586

### CODE REVIEW

**Review by wsmoses — APPROVED** [review]
Fix logb comment, otherwise LGTM

File: enzyme/Enzyme/AdjointGenerator.h [hunk 1]
```diff
@@ -9239,6 +9239,349 @@ class AdjointGenerator
...
+            eraseIfUnused(*orig);
+
+          return;
+        }
+        case DerivativeMode::ReverseModePrimal:;
+          return;
+        }
+      }
+
+      if (funcName == "logb" || funcName == "logbf" || funcName == "logbl") {
```
[the remark this answers is not in the document]
Comment (tgymnich) [reply enzyme/Enzyme/AdjointGenerator.h]: done

**Review by wsmoses — APPROVED** [review]
Add vector tests and address the remaining comment otherwise LGTM

File: enzyme/Enzyme/TypeAnalysis/TypeAnalysis.cpp [hunk 2]
```diff
@@ -91,12 +91,21 @@ const std::map<std::string, llvm::Intrinsic::ID> LIBM_FUNCTIONS = {
     {"expm1", Intrinsic::not_intrinsic},
     {"log1p", Intrinsic::not_intrinsic},
     {"log2", Intrinsic::log2},
-    {"logb", Intrinsic::not_intrinsic},
```
[the remark this answers is not in the document]
Comment (tgymnich) [reply enzyme/Enzyme/TypeAnalysis/TypeAnalysis.cpp:94]: fixed

**Pull request merged**